evm-ext/src/native_impl.rs

```diff
@@ -157,6 +175,32 @@ fn evm_transact_inner(
         })
         .build();
 
+    // Modify the post-execution handler to extract transfer events.
+    evm.handler.post_execution.output = Arc::new(|evm_ctx, result| {
+        let transfers = evm_ctx
+            .evm
+            .journaled_state
+            .journal
+            .iter()
+            .flat_map(|entries| {
+                entries.iter().filter_map(|entry| {
+                    if let revm::JournalEntry::BalanceTransfer { from, to, balance } = entry {
+                        Some(EthTransfer {
+                            from: from.to_move_address(),
+                            to: to.to_move_address(),
+                            amount: *balance,
+                        })
+                    } else {
+                        None
+                    }
+                })
+            });
+        for t in transfers {
+            transfers_log.borrow_mut().push(t);
+        }
```

Comment:
It seems we can just directly `evm_native_ctx.transfers_log.append_transfers(transfers)` here after collecting them into a vec, or at least the compiler doesn't complain. Am I missing some reason the intermediary refcell is needed?

Reply:
Good catch!

This PR went through a couple iterations and in an earlier version the logger itself was mutable, so it couldn't go here and the intermediary was needed. But that mutability ended up being a problem somewhere else, so I changes the type signature to immutable and now there is no reason for the intermediary. I'll simplify it.

Reply:
Fixed in https://github.com/MovedNetwork/op-move/pull/319/commits/4b1845a594e661cbbc0db45dc6ce5dd181271738